alfalfa_worker/jobs/modelica/step_run.py

```diff
@@ -110,22 +105,27 @@ def get_sim_time(self) -> datetime:
     def step(self):
         # u represents simulation input values
         u = self.default_input.copy()
-        # look in the database for current write arrays
-        # for each write array there is an array of controller
+
+        # look in the database for current writearrays which has an array of controller
         # input values, the first element in the array with a value
         # is what should be applied to the simulation according to Project Haystack
-        # convention
-        for array in self.mongo_db_write_arrays.find({"siteRef": self.run.id}):
-            _id = array.get('_id')
-            for val in array.get('val'):
+        # convention. If there is no value in the array, then it will not be passed to the
+        # simulation.
+        for array in WriteArray.objects(siteRef=self.site.ref_id):
```

Comment:
i really wanted to abstract this out, but the way that modelica handles this is different than openstudio.